### DISCUSSION TITLE
Unable to create the key

### DISCUSSION THREAD

**Discussion by bortoli99** [laravel-notification-channels/webpush#206] [Q&A]
php artisan webpush:vapid

   RuntimeException 

  Unable to create the key

  at vendor\web-token\jwt-library\Core\Util\ECKey.php:96
     92▕             'curve_name' => self::getOpensslCurveName($curve),
     93▕             'private_key_type' => OPENSSL_KEYTYPE_EC,
     94▕         ]);
     95▕         if ($key === false) {
  ➜  96▕             throw new RuntimeException('Unable to create the key');
     97▕         }
     98▕         $result = openssl_pkey_export($key, $out);
     99▕         if ($result === false) {
    100▕             throw new RuntimeException('Unable to create the key');

  1   vendor\web-token\jwt-library\Core\Util\ECKey.php:70
      Jose\Component\Core\Util\ECKey::createECKeyUsingOpenSSL("P-256")

  2   vendor\web-token\jwt-library\KeyManagement\JWKFactory.php:72
      Jose\Component\Core\Util\ECKey::createECKey("P-256", [])

**Comment by Somenamefy** [comment]
Same here. Can't generate key.

**Comment by dlabsnl** [comment]
https://slproweb.com/products/Win32OpenSSL.html

Do you have this installed?

**Comment by einenlum** [comment]
The problem for me was related to PHP Herd lite. It seems like it doesn't use the right path for ssl configuration (`/ssl`). I managed to fix this with:

`OPENSSL_CONF=/etc/ssl/openssl.cnf php artisan webpush:vapid`
